libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/SharedApp.kt

```diff
@@ -2,24 +2,56 @@ package com.mapbox.navigation.ui.app.internal
 
 import android.content.Context
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.internal.extensions.attachCreated
 import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.ui.app.internal.controller.AudioGuidanceStateController
+import com.mapbox.navigation.ui.app.internal.controller.CameraStateController
+import com.mapbox.navigation.ui.app.internal.controller.DestinationStateController
+import com.mapbox.navigation.ui.app.internal.controller.LocationStateController
+import com.mapbox.navigation.ui.app.internal.controller.NavigationStateController
+import com.mapbox.navigation.ui.app.internal.controller.RoutesStateController
+import com.mapbox.navigation.ui.app.internal.controller.TripSessionStarterStateController
 import com.mapbox.navigation.ui.utils.internal.datastore.NavigationDataStoreOwner
 import com.mapbox.navigation.ui.voice.internal.MapboxAudioGuidance
 import com.mapbox.navigation.ui.voice.internal.impl.MapboxAudioGuidanceImpl
 
-@ExperimentalPreviewMapboxNavigationAPI
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 object SharedApp {
     private var isSetup = false
 
+    val store = Store()
+    val state get() = store.state.value
+
+    /**
+     * These classes are accessible through MapboxNavigationApp.getObserver(..)
+     */
+    val navigationStateViewModel = NavigationStateController(store)
+    val locationViewModel = LocationStateController(store)
+    val tripSessionStarterViewModel = TripSessionStarterStateController(store)
+    val audioGuidanceViewModel = AudioGuidanceStateController(store)
+    val cameraViewModel = CameraStateController(store)
+    val destinationViewModel = DestinationStateController(store)
+    val routesViewModel = RoutesStateController(store)
```

Comment:
NIT:
How about
`val navigationStateController = NavigationStateController(store)`
Same for all the others.

Reply:
Yeah, this will rename everything downstream. I can go for it now